### PR TITLE
[Search] Add a sample to show case how to use SessionId to create sticky sessions

### LINKED ISSUE #39196: [BUG] Preview SearchModelFactory.SearchResults<T> does not allow mocking of Answers or Captions
### Library name and version

Azure.Search.Documents v11.5.0-beta.5

### Describe the bug

Preview version of SearchModelFactory.SearchResults<T> does not take in parameters for Answers, Captions, SemanticPartialResponseReason, and SemanticPartialResponseType

### Expected behavior

Answers, Captions, SemanticPartialResponseReason, and SemanticPartialResponseType should be passable as parameters to create SearchResults instances for mocking that include those properties

### Actual behavior

Answers, Captions, SemanticPartialResponseReason, and SemanticPartialResponseType are not passable as parameters, causing those properties to become unmockable when part of SearchResults

### Reproduction Steps

Attempt to mock SearchAsync like the following:

```
var clientMock = new Mock<SearchClient>();
var responseMock = new Mock<Response>();

var resultMock = SearchModelFactory.SearchResult(new SearchDocument() { { "metadata_storage_name", null } }, null, null);
var resultsMock = SearchModelFactory.SearchResults(new[] { resultMock }, 1, null, null, responseMock.Object);

clientMock.Setup(x => x.SearchAsync<SearchDocument>(It.IsAny<string>(), It.IsAny<SearchOptions>(), It.IsAny<CancellationToken>()))
    .ReturnsAsync(Response.FromValue(resultsMock, responseMock.Object));
```

### Environment

_No response_

### LINKED ISSUE #41413: Add a sample to show case how to use SessionId to create sticky sessions in Search SDK
Search team recently reached with a customer reported issue complaining about duplicate / incorrect results while paging using the filter. It was determined the root cause of the issue was that service expects sticky session to support page query in a multi-replica service.
 
The way to mitigate this is to use the session id feature [https://learn.microsoft.com/en-us/azure/search/index-similarity-and-scoring#scoring-statistics-and-sticky-sessions](https://learn.microsoft.com/en-us/azure/search/index-similarity-and-scoring)

Given the issue we want to add a sample to educate the customers of this gap.

### PR DESCRIPTION
**Pull request by ShivangiReja**
This PR includes the following:
* A sample to show case how to use SessionId to create sticky sessions. Fixes: https://github.com/Azure/azure-sdk-for-net/issues/41413
* Missing parameters in `SearchModelFactory.SearchResults<T>` and `SearchModelFactory.SearchResult<T>` for mocking. Fixes: https://github.com/Azure/azure-sdk-for-net/issues/39196

### CODE REVIEW

File: sdk/search/Azure.Search.Documents/src/Models/SearchResults.cs
Comment (heaths) [inline sdk/search/Azure.Search.Documents/src/Models/SearchResults.cs:287]: Same comment here as for `SearchResult`.

File: sdk/search/Azure.Search.Documents/tests/Samples/Sample01_HelloWorld.cs
Comment (heaths) [inline sdk/search/Azure.Search.Documents/tests/Samples/Sample01_HelloWorld.cs:491]: @tg-msft @KrzysztofCwalina are we switching to async samples now? I've always thought we should, but to my knowledge we've always opted first and foremost for sync for "ease".

File: sdk/search/Azure.Search.Documents/tests/Samples/Sample01_HelloWorld.cs
Comment (heaths) [inline sdk/search/Azure.Search.Documents/tests/Samples/Sample01_HelloWorld.cs:500]: Why are we enumerating pages? This is not typically needed and complicates the example. Just `await foreach` over the whole collection unless you really need to query pages. Is it only to get page count? I wouldn't think most people would care. Our samples should show the mainline cases unless specifically for non-mainline cases, which as a "Hello world" sample doesn't seem appropriate here.

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs
Comment (heaths) [inline sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:3472]: Where did these come from? I don't see the changes in the PR. If from before, this means the checks would've failed prior.

**Review by heaths — CHANGES_REQUESTED** [review]

File: sdk/search/Azure.Search.Documents/src/Models/SearchResult.cs
Comment (ShivangiReja) [reply sdk/search/Azure.Search.Documents/src/Models/SearchResult.cs]: Removed the ctors.

File: sdk/search/Azure.Search.Documents/tests/Samples/Sample01_HelloWorld.cs
Comment (ShivangiReja) [reply sdk/search/Azure.Search.Documents/tests/Samples/Sample01_HelloWorld.cs]: Most of samples are async in Search:
* [Index Operations](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/search/Azure.Search.Documents/samples/Sample03_Index.md)
* [Indexing Documents](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/search/Azure.Search.Documents/samples/Sample05_IndexingDocuments.md)
* [All the Vector Search samples](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/search/Azure.Search.Documents/samples/Sample07_VectorSearch.md)
* [Semantic Search](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/search/Azure.Search.Documents/samples/Sample08_SemanticSearch.md)

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs
Comment (ShivangiReja) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:3472]: We have added `[EditorBrowsable(EditorBrowsableState.Never)]` into the [SearchResults.cs](https://github.com/Azure/azure-sdk-for-net/pull/41694/files#diff-5c0e533bf06ba67f92e99afce2b578bb80514e7aa77ca02cc58c59651787ded1) file in this PR and added a new mocking constructor. This was added as the SemanticSearchResults property was missing in the existing mocking constructor.

Comment (heaths) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:3472]: Why are you doing static factory methods on the types themselves, though? Our [guidelines state](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-mocking-factory-builder):

> ✅ DO provide factory or builder for constructing model graphs returned from virtual service methods.

> 
> Model types shouldn’t have public constructors. Instances of the model are typically returned from the client library, and are not constructed by the consumer of the library. Mock implementations need to create instances of model types. Implement a static class called <service>ModelFactory in the same namespace as the model types:

Or are you just continuing a trend this library already seems to have? Has it GA'd this way through architect approval?

Comment (ShivangiReja) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:3472]: @heaths - The static factory method is not on the type itself. [SearchModelFactory](https://github.com/Azure/azure-sdk-for-net/blob/588852238962ab0f08f8bc2143ea7e50bfc90577/sdk/search/Azure.Search.Documents/src/Models/SearchResults.cs#L433) is a static partial class created inside the `SearchResults.cs` file, and all the factory methods are inside the [SearchModelFactory](https://github.com/Azure/azure-sdk-for-net/blob/588852238962ab0f08f8bc2143ea7e50bfc90577/sdk/search/Azure.Search.Documents/src/Models/SearchResults.cs#L433-L477) type.

Comment (heaths) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:3472]: I see now. That's really confusing to reviewers and maintainers and requires suppressing a warning not to put multiple types in a single source file. I think we should avoid this pattern in the future.

**Review by heaths — APPROVED** [review]

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs
Comment (ShivangiReja) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:3472]: I completely agree with that. I have been following to the existing pattern, but I will create a separate PR for this cleanup.

**Pull request merged**
